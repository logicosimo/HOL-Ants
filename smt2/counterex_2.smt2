(declare-datatype Position ((P0) (P1) (P2) (P3) (P4)))
(declare-const a Int)
(declare-const a_1 Int)
(declare-const a_2 Int)
(declare-const a_3 Int)
(declare-const a_4 Int)
(declare-const a_5 Int)
(declare-const c Int)
(declare-const c_1 Int)
(declare-const c_2 Int)
(declare-const dir Bool)
(declare-const dir_1 Bool)
(declare-const dir_2 Bool)
(declare-const dir_3 Bool)
(declare-const dir_4 Bool)
(declare-const dir_5 Bool)
(declare-const pos Position)
(declare-const pos_1 Position)
(declare-const pos_2 Position)
(declare-const pos_3 Position)
(declare-const pos_4 Position)
(declare-const pos_5 Position)
(assert (>= a 0))
(assert (>= a_1 0))
(assert (>= a_2 0))
(assert (>= a_3 0))
(assert (>= a_4 0))
(assert (>= a_5 0))
(assert (>= c 0))
(assert (>= c_1 0))
(assert (>= c_2 0))
(assert
 (=>
  (and (= a_2 (+ a (ite (= pos P1) 1 0) (ite (= pos_1 P1) 1 0)))
   (= c_1 (+ c (ite (= pos P3) 1 0) (ite (= pos_1 P3) 1 0)))
   (= a_3 (+ a_1 (ite (= pos P2) 1 0) (ite (= pos_1 P2) 1 0)))
   (=> (= pos P1) (and (= pos_2 (ite dir P4 P0)) (iff dir_2 dir)))
   (=> (= pos P2) (and (= pos_2 (ite dir P3 P0)) (iff dir_2 dir)))
   (=> (= pos P3) (and (= pos_2 (ite dir P4 P2)) (iff dir_2 dir)))
   (=> (= pos P0) (and (or (= pos_2 P1) (= pos_2 P2)) dir_2))
   (=> (and (= pos P0) (= pos_2 P1)) (<= a_1 a))
   (=> (and (= pos P0) (= pos_2 P2)) (<= a a_1))
   (=> (= pos P4) (and (or (= pos_2 P1) (= pos_2 P3)) (not dir_2)))
   (=> (and (= pos P4) (= pos_2 P1)) (<= c a))
   (=> (and (= pos P4) (= pos_2 P3)) (<= a c))
   (=> (= pos_1 P1) (and (= pos_3 (ite dir_1 P4 P0)) (iff dir_3 dir_1)))
   (=> (= pos_1 P2) (and (= pos_3 (ite dir_1 P3 P0)) (iff dir_3 dir_1)))
   (=> (= pos_1 P3) (and (= pos_3 (ite dir_1 P4 P2)) (iff dir_3 dir_1)))
   (=> (= pos_1 P0) (and (or (= pos_3 P1) (= pos_3 P2)) dir_3))
   (=> (and (= pos_1 P0) (= pos_3 P1)) (<= a_1 a))
   (=> (and (= pos_1 P0) (= pos_3 P2)) (<= a a_1))
   (=> (= pos_1 P4) (and (or (= pos_3 P1) (= pos_3 P3)) (not dir_3)))
   (=> (and (= pos_1 P4) (= pos_3 P1)) (<= c a))
   (=> (and (= pos_1 P4) (= pos_3 P3)) (<= a c))
   (= a_4 (+ a_2 (ite (= pos_2 P1) 1 0) (ite (= pos_3 P1) 1 0)))
   (= c_2 (+ c_1 (ite (= pos_2 P3) 1 0) (ite (= pos_3 P3) 1 0)))
   (= a_5 (+ a_3 (ite (= pos_2 P2) 1 0) (ite (= pos_3 P2) 1 0)))
   (=> (= pos_2 P1) (and (= pos_4 (ite dir_2 P4 P0)) (iff dir_4 dir_2)))
   (=> (= pos_2 P2) (and (= pos_4 (ite dir_2 P3 P0)) (iff dir_4 dir_2)))
   (=> (= pos_2 P3) (and (= pos_4 (ite dir_2 P4 P2)) (iff dir_4 dir_2)))
   (=> (= pos_2 P0) (and (or (= pos_4 P1) (= pos_4 P2)) dir_4))
   (=> (and (= pos_2 P0) (= pos_4 P1)) (<= a_3 a_2))
   (=> (and (= pos_2 P0) (= pos_4 P2)) (<= a_2 a_3))
   (=> (= pos_2 P4) (and (or (= pos_4 P1) (= pos_4 P3)) (not dir_4)))
   (=> (and (= pos_2 P4) (= pos_4 P1)) (<= c_1 a_2))
   (=> (and (= pos_2 P4) (= pos_4 P3)) (<= a_2 c_1))
   (=> (= pos_3 P1) (and (= pos_5 (ite dir_3 P4 P0)) (iff dir_5 dir_3)))
   (=> (= pos_3 P2) (and (= pos_5 (ite dir_3 P3 P0)) (iff dir_5 dir_3)))
   (=> (= pos_3 P3) (and (= pos_5 (ite dir_3 P4 P2)) (iff dir_5 dir_3)))
   (=> (= pos_3 P0) (and (or (= pos_5 P1) (= pos_5 P2)) dir_5))
   (=> (and (= pos_3 P0) (= pos_5 P1)) (<= a_3 a_2))
   (=> (and (= pos_3 P0) (= pos_5 P2)) (<= a_2 a_3))
   (=> (= pos_3 P4) (and (or (= pos_5 P1) (= pos_5 P3)) (not dir_5)))
   (=> (and (= pos_3 P4) (= pos_5 P1)) (<= c_1 a_2))
   (=> (and (= pos_3 P4) (= pos_5 P3)) (<= a_2 c_1))
   (> a (ite (<= a_1 c) c a_1)) (> a_2 (ite (<= a_3 c_1) c_1 a_3)))
  (> a_4 (ite (<= a_5 c_2) c_2 a_5))))
(check-sat)
(get-value
 (a a_1 a_2 a_3 a_4 a_5 c c_1 c_2 dir dir_1 dir_2 dir_3 dir_4 dir_5 pos pos_1
  pos_2 pos_3 pos_4 pos_5))
