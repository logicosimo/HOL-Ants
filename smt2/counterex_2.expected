sat
((a 0)
 (a_1 0)
 (a_2 1)
 (a_3 0)
 (a_4 0)
 (a_5 0)
 (c 0)
 (c_1 0)
 (c_2 0)
 (dir false)
 (dir_1 false)
 (dir_2 false)
 (dir_3 false)
 (dir_4 false)
 (dir_5 false)
 (pos P2)
 (pos_1 P0)
 (pos_2 P4)
 (pos_3 P4)
 (pos_4 P0)
 (pos_5 P3))
